File: 2875.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    //cin.tie(NULL);
    ios_base::sync_with_stdio(0);
    freopen("input.txt", "r", stdin);

    int N, M, K;
    cin >> N >> M >> K;

    while(K) {
        if(2*M > N) {
            M--;
            K--;
        } else if(2*M < N) {
            N--;
            K--;
        } else {
            M--;
            K--;
        }
    }

    if(2*M > N) {
        cout << N/2;
    } else {
        cout << M;
    }
    
	return 0;
}

File: 2875.py
import sys
sys.stdin = open("input.txt", 'r')
input = sys.stdin.readline

N, M, K = map(int, input().split())
cnt_team = min(N//2, M)
rest = N+M-(cnt_team*3)
rest -= K
rest //= 3
if rest < 0:
    cnt_team += rest
print(cnt_team)
